icu4c/source/common/cjmlbe.cpp

```diff
@@ -0,0 +1,462 @@
+    }
+    if (b3 != UnicodeString(INVALID)) {
+        featureList->adoptElement(concatUnicodeString("UB3:", b3, status), status);
+    }
+    if (b4 != UnicodeString(INVALID)) {
+        featureList->adoptElement(concatUnicodeString("UB4:", b4, status), status);
+    }
+    if (b5 != UnicodeString(INVALID)) {
+        featureList->adoptElement(concatUnicodeString("UB5:", b5, status), status);
+    }
+    if (b6 != UnicodeString(INVALID)) {
+        featureList->adoptElement(concatUnicodeString("UB6:", b6, status), status);
+    }
+    if (b2 != UnicodeString(INVALID) && b3 != UnicodeString(INVALID)) {
+        featureList->adoptElement(concatUnicodeString("BB1:", UnicodeString(b2).append(b3), status), status);
+    }
+    if (b3 != UnicodeString(INVALID) && b4 != UnicodeString(INVALID)) {
+        featureList->adoptElement(concatUnicodeString("BB2:", UnicodeString(b3).append(b4), status), status);
+    }
+    if (b4 != UnicodeString(INVALID) && b5 != UnicodeString(INVALID)) {
+        featureList->adoptElement(concatUnicodeString("BB3:", UnicodeString(b4).append(b5), status), status);
+    }
+    if (b1 != UnicodeString(INVALID) && b2 != UnicodeString(INVALID) && b3 != UnicodeString(INVALID)) {
+        featureList->adoptElement(concatUnicodeString("TB1:", UnicodeString(b1).append(b2).append(b3), status), status);
+    }
+    if (b2 != UnicodeString(INVALID) && b3 != UnicodeString(INVALID) && b4 != UnicodeString(INVALID)) {
+        featureList->adoptElement(concatUnicodeString("TB2:", UnicodeString(b2).append(b3).append(b4), status), status);
+    }
+    if (b3 != UnicodeString(INVALID) && b4 != UnicodeString(INVALID) && b5 != UnicodeString(INVALID)) {
+        featureList->adoptElement(concatUnicodeString("TB3:", UnicodeString(b3).append(b4).append(b5), status), status);
+    }
+    if (b4 != UnicodeString(INVALID) && b5 != UnicodeString(INVALID) && b6 != UnicodeString(INVALID)) {
+        featureList->adoptElement(concatUnicodeString("TB4:", UnicodeString(b4).append(b5).append(b6), status), status);
+    }
+
+    int32_t score = negativeSum;
+    for (int32_t j = 0; j < featureList->size(); j++) {
+        UnicodeString* key = (UnicodeString*)featureList->elementAt(j);
+        if(fModel.containsKey(*key)) {
+            score += (2 * fModel.geti(*key));
+        }
+    }
+
+    if (score > 0) {
+        boundary.addElement(index, status);
+        numBreaks++;
+    }
+
+    featureList->removeAllElements();
+    delete featureList;
+}
+
+void CjMLBreakEngine::initElementList(UnicodeString& inString, UVector* elementList, UErrorCode& status) const {
+    UChar32 w1, w2, w3, w4, w5, w6;
+    UnicodeString b1, b2, b3, b4, b5, b6;
+    w1 = INVALID;
+    w2 = INVALID;
+    w3 = inString.char32At(0);
+    w4 = inString.char32At(1);
+    w5 = 2 < inString.length() ? inString.char32At(2) : INVALID;
+    w6 = 3 < inString.length() ? inString.char32At(3) : INVALID;
+
+    b1 = UnicodeString(INVALID);
```

Comment:
How about remove the line "    UnicodeString b1, b2, b3, b4, b5, b6;"
and change them to 

    const UnicodeString b1(INVALID);
    const UnicodeString b2(b1);
    const UnicodeString b3(getUnicodeBlockIndex(w3, status));
    const UnicodeString b4(getUnicodeBlockIndex(w4, status));
    const UnicodeString b5(getUnicodeBlockIndex(w5, status));
    const UnicodeString b6(getUnicodeBlockIndex(w6, status));

Reply:
Done. Thanks